cmd/tetra/explain/explain.go

```diff
@@ -0,0 +1,519 @@
+
+Supported resources:
+  tracingpolicy, tp         - TracingPolicy CRD
+  tracingpolicynamespaced   - TracingPolicyNamespaced CRD  
+  podinfo                   - PodInfo CRD
+
+Path syntax supports both dot and bracket notation:
+  tracingpolicy.spec.kprobes     (dot notation)
+  tracingpolicy[spec][kprobes]   (bracket notation)
+  tracingpolicy[spec].kprobes    (mixed notation)
+
+Examples:
+  tetra explain tracingpolicy
+  tetra explain tracingpolicy.spec
+  tetra explain tracingpolicy.spec.kprobes
+  tetra explain tracingpolicy[spec][kprobes]
+  tetra explain tp.spec.calls
+  tetra explain podinfo
+  tetra explain --list
+  tetra explain tracingpolicy.spec --recursive
+  tetra explain tracingpolicy.spec -o json
+`,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if listMode {
+				return nil // --list doesn't require args
+			}
+			return cobra.ExactArgs(1)(cmd, args)
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			if listMode {
+				listResources(cmd.OutOrStdout())
+				return
+			}
+
+			if err := runExplain(cmd.OutOrStdout(), args[0]); err != nil {
+				fmt.Fprintf(cmd.OutOrStderr(), "Error: %v\n", err)
+				os.Exit(1)
+			}
+		},
+	}
+
+	cmd.Flags().StringVarP(&outputFormat, "output", "o", "", "Output format. One of: json|yaml")
+	cmd.Flags().BoolVarP(&recursive, "recursive", "R", false, "Print the fields of fields recursively")
+	cmd.Flags().BoolVar(&showExample, "example", false, "Show example usage")
+	cmd.Flags().BoolVar(&listMode, "list", false, "List all supported resources")
+	cmd.Flags().StringVar(&apiVersion, "api-version", "", "Specify the API version to use")
+
+	return cmd
+}
+
+var supportedResources = map[string]resourceInfo{
+	"tracingpolicy": {
+		kind:    "TracingPolicy",
+		aliases: []string{"tp"},
+		getCRD: func() *apiextensionsv1.CustomResourceDefinition {
+			return &client.TracingPolicyCRD.Definition
+		},
+	},
+	"tracingpolicynamespaced": {
+		kind:    "TracingPolicyNamespaced",
+		aliases: []string{"tpns"},
+		getCRD: func() *apiextensionsv1.CustomResourceDefinition {
+			return &client.TracingPolicyNamespacedCRD.Definition
+		},
+	},
+	"podinfo": {
+		kind:    "PodInfo",
+		aliases: []string{},
+		getCRD: func() *apiextensionsv1.CustomResourceDefinition {
+			return &client.PodInfoCRD.Definition
+		},
+	},
+}
+
+func listResources(w io.Writer) {
+	tw := tabwriter.NewWriter(w, 0, 0, 2, ' ', 0)
+	fmt.Fprintln(tw, "NAME\tSHORTNAMES\tKIND")
+
+	var resources []string
+	for name := range supportedResources {
+		resources = append(resources, name)
+	}
+	sort.Strings(resources)
+
+	for _, name := range resources {
+		info := supportedResources[name]
+		shortNames := strings.Join(info.aliases, ",")
+		fmt.Fprintf(tw, "%s\t%s\t%s\n", name, shortNames, info.kind)
+	}
+	tw.Flush()
+}
+
+// convertBracketsToDots converts bracket notation to dot notation using regex
+func convertBracketsToDots(path string) string {
+	path = strings.ReplaceAll(path, "][", ".")
```

Comment:
Since after this we end up with, eg: `tracingpolicy[spec][uprobes][tags]` -> `tracingpolicy[spec.uprobes.tags]`, is it really worth it all the regex-related logic? After all, we can just safely replace `[` with `.` and drop last `]`, right?
Or is there an edge case i am not considering?